modular_splurt/code/datums/traits/trait_actions.dm

```diff
@@ -0,0 +1,343 @@
+		H.update_body()
+		H.update_body_parts()
+		H.update_size(get_size(H) - 0.5)
+		if(B)
+			B.color = "#[old_features["breasts_color"]]"
+			B.update()
+		if(H.has_penis())
+			P.shape = old_features["cock_shape"]
+			P.color = "#[old_features["cock_color"]]"
+			P.update()
+			P.modify_size(-6)
+		if(H.has_vagina())
+			V.shape = old_features["vag_shape"]
+			V.color = "#[old_features["vag_color"]]"
+			V.update()
+			V.update_size()
+	transformed = !transformed
+
+/datum/action/werewolf/Grant()// on grant sets some variables
+	. = ..()
+	var/mob/living/carbon/human/H = owner
+	old_features = H.dna.features.Copy()
+	old_features["species"] = H.dna.species.type
+	old_features["size"] = get_size(H)
+	old_features["bark"] = H.vocal_bark_id
+
+
+/datum/action/gargoyle/transform
+	name = "Transform"
+	desc = "Transform into a statue, regaining energy in the process. Additionally, you will slowly heal while in statue form."
+	icon_icon = 'icons/mob/actions/actions_changeling.dmi'
+	button_icon_state = "ling_camouflage"
+	var/obj/structure/statue/gargoyle/current = null
+
+
+/datum/action/gargoyle/transform/Trigger()
+	.=..()
+	var/mob/living/carbon/human/H = owner
+	var/datum/quirk/gargoyle/T = locate() in H.roundstart_quirks
+	if(!T.cooldown)
+		if(!T.transformed)
+			if(!isturf(H.loc))
+				return 0
+			var/obj/structure/statue/gargoyle/S = new(H.loc, H)
+			S.name = "statue of [H.name]"
+			H.bleedsuppress = 1
+			S.copy_overlays(H)
+			var/newcolor = list(rgb(77,77,77), rgb(150,150,150), rgb(28,28,28), rgb(0,0,0))
+			S.add_atom_colour(newcolor, FIXED_COLOUR_PRIORITY)
+			current = S
+			T.transformed = 1
+			T.cooldown = 30
+			T.paused = 0
+			S.dir = H.dir
+			return 1
+		else
+			qdel(current)
+			T.transformed = 0
+			T.cooldown = 30
+			T.paused = 0
+			H.visible_message("<span class='warning'>[H]'s skin rapidly softens, returning them to normal!</span>", "<span class='userdanger'>Your skin softens, freeing your movement once more!</span>")
+	else
+		to_chat(H, "<span class='warning'>You have transformed too recently; you cannot yet transform again!</span>")
+		return 0
+
+//
+// Quirk: Gargoyle
+//
+
+/datum/action/gargoyle/check
+	name = "Check"
+	desc = "Check your current energy levels."
+	icon_icon = 'icons/mob/actions/actions_clockcult.dmi'
+	button_icon_state = "Linked Vanguard"
+
+/datum/action/gargoyle/check/Trigger()
+	.=..()
+	var/mob/living/carbon/human/H = owner
+	var/datum/quirk/gargoyle/T = locate() in H.roundstart_quirks
+	to_chat(H, "<span class='warning'>You have [T.energy]/100 energy remaining!</span>")
+
+/datum/action/gargoyle/pause
+	name = "Preserve"
+	desc = "Become near-motionless, thusly conserving your energy until you move from your current tile. Note, you will lose a chunk of energy when you inevitably move from your current position, so you cannot abuse this!"
+	icon_icon = 'icons/mob/actions/actions_flightsuit.dmi'
+	button_icon_state = "flightsuit_lock"
+
+datum/action/gargoyle/pause/Trigger()
```

Comment:
```suggestion
/datum/action/gargoyle/pause/Trigger()
```

Reply:
Surprisingly enough, this exists in the original file.